implementations/rust/ockam/ockam_command/src/node/util.rs

```diff
@@ -90,12 +91,12 @@ pub async fn start_embedded_node_with_vault_and_identity(
             None,
         ),
         NodeManagerTransportOptions::new(
-            (
-                TransportType::Tcp,
-                TransportMode::Listen,
-                listened_worker_address,
-                socket_addr.to_string(),
```

Comment:
I found naming for the arguments a bit confusing. I derived the names of the fields for the new `ApiTransport` struct from names of variables that were passed into a former tuple. However, in this bit of code `socket_addr` is the 4th argument, whilst in `implementations/rust/ockam/ockam_api/src/nodes/service/transport.rs:27` it is the 3rd one. I decided to save the order in all the source files despite the naming discrepancies. However, it would be awesome to get some comment on this matter.

Reply:
This looks like a bug. It's harmless because we currently do nothing with that data other than displaying it as some command's output. A couple of things on this:

- We should also fix the typo: `listener` instead of `listened`
- Would it be possible to store both addresses as `Address` type and not use `String`?

Reply:
Okay, got it! I will change the order of the arguments and fix the type then.
In the struct `ApiTransport` the field `socket_address` is already of the type `Address`, not `String`, although I could try to change it to `SockAddr`. Should we change `worker_address` type to `Address` too?

Reply:
Yes sorry, I think I didn't express that part correctly. It should be:
- socket_address: SocketAddr
- worker_address: Address

After that change, the arguments order should be pretty straightforward to fix.

Reply:
What do you think keeping socket_address as `Address` type? It seems to be used in `implementations/rust/ockam/ockam_transport_tcp/src/transport.rs:196`, which receives the `Address` object. Although we still need `From<SocketAddr>` trait implementation, I added it to `ockam_core`.

Reply:
What do you think about using `Address` instead of `SocketAddr` for `socket_address` field for now? I tried to implement change it, but it requires parsing from string in different parts of the codebase, which can throw an error. I also stumbled upon new errors during the bats run.

For now I implemented `From<SocketAddr>` for `Address` and convert SocketAddr to Address in `ockam_commands` right away.

Reply:
That's incorrect I'm afraid. There is no direct conversion between `SocketAddr` <--> `Address`. The only way to do that would be to use a `SocketAddr` to establish a transport connection (TCP for example), which returns a worker `Address`.

That's why `socket_addr` must be of type `SocketAddr`.